cshared.go

```diff
@@ -133,47 +136,79 @@ func FLBPluginInputCallback(data *unsafe.Pointer, csize *C.size_t) int {
 	var err error
 	once.Do(func() {
 		runCtx, runCancel = context.WithCancel(context.Background())
-		theChannel = make(chan Message)
+		// we need to configure this part....
+		theChannel = make(chan Message, 300000)
+		// do we need to buffer this part???
+		cbuf := make(chan Message, 16)
+
 		go func() {
-			err = theInput.Collect(runCtx, theChannel)
+			err = theInput.Collect(runCtx, cbuf)
 		}()
+		go func(cbuf chan Message) {
+			t := time.NewTicker(1 * time.Second)
+			for {
+				buflock.Lock()
+				select {
+				case msg, ok := <-cbuf:
+					if !ok {
+						continue
+					}
+					buflock.Unlock()
+					theChannel <- msg
+					buflock.Lock()
+				case <-t.C:
+					buflock.Unlock()
```

Comment:
This is smelly, do we need this? I am not sure if following the logic here.

Reply:
I have added comments on why the lock is being used.